build/patches/Add-a-flag-to-always-view-desktop-site-for-all-websites.patch

```diff
@@ -0,0 +1,471 @@
+diff --git a/chrome/android/java/res/xml/useragent_preferences.xml b/chrome/android/java/res/xml/useragent_preferences.xml
+new file mode 100644
+--- /dev/null
++++ b/chrome/android/java/res/xml/useragent_preferences.xml
+@@ -0,0 +1,30 @@
++<?xml version="1.0" encoding="utf-8"?>
++<!--
++    This file is part of Bromite.
++
++    Bromite is free software: you can redistribute it and/or modify
++    it under the terms of the GNU General Public License as published by
++    the Free Software Foundation, either version 3 of the License, or
++    (at your option) any later version.
++
++    Bromite is distributed in the hope that it will be useful,
++    but WITHOUT ANY WARRANTY; without even the implied warranty of
++    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
++    GNU General Public License for more details.
++
++    You should have received a copy of the GNU General Public License
++    along with Bromite. If not, see <https://www.gnu.org/licenses/>.
++-->
++
++<!-- Layout used by the UserAgentPreferences. -->
++
++<PreferenceScreen
++    xmlns:android="http://schemas.android.com/apk/res/android"
++    xmlns:app="http://schemas.android.com/apk/res-auto">
++
++    <org.chromium.components.browser_ui.settings.ChromeSwitchPreference
++        android:key="desktop_mode_switch"
++        android:summaryOn="@string/option_desktop_flag_on"
++        android:summaryOff="@string/option_desktop_flag_off" />
++
++</PreferenceScreen>
+diff --git a/chrome/android/java/src/org/chromium/chrome/browser/app/ChromeActivity.java b/chrome/android/java/src/org/chromium/chrome/browser/app/ChromeActivity.java
+--- a/chrome/android/java/src/org/chromium/chrome/browser/app/ChromeActivity.java
++++ b/chrome/android/java/src/org/chromium/chrome/browser/app/ChromeActivity.java
+@@ -202,6 +202,9 @@ import org.chromium.ui.modaldialog.ModalDialogManager;
+ import org.chromium.ui.widget.Toast;
+ import org.chromium.url.Origin;
+ import org.chromium.webapk.lib.client.WebApkNavigationClient;
++import org.chromium.chrome.browser.tabmodel.TabWindowManager;
++import org.chromium.chrome.browser.preferences.ChromePreferenceKeys;
++import org.chromium.chrome.browser.preferences.SharedPreferencesManager;
+ 
+ import java.util.ArrayList;
+ import java.util.List;
+@@ -1987,11 +1990,19 @@ public abstract class ChromeActivity<C extends ChromeActivityComponent>
+         } else if (id == R.id.view_source_id) {
+             currentTab.getWebContents().getNavigationController().loadUrl(new LoadUrlParams("view-source:"+currentTab.getUrlString()));
+         } else if (id == R.id.request_desktop_site_id || id == R.id.request_desktop_site_check_id) {
+-            final boolean reloadOnChange = !currentTab.isNativePage();
+             final boolean usingDesktopUserAgent =
+                     currentTab.getWebContents().getNavigationController().getUseDesktopUserAgent();
+-            currentTab.getWebContents().getNavigationController().setUseDesktopUserAgent(
```

Comment:
`setUseDesktopUserAgent` is used only here; it is OK to leave it behind (unless it causes build failure, will see) but I think we should preserve the `reloadOnChange` behavior. I am assuming that it is an optimisation used because native pages do not need to reload when agent changes.

No need to pass an extra parameter here, it can be checked in the plumbing of `SetOverrideUserAgentForAllTabs`/`SetOverrideUserAgent`. What do you think?

Reply:
> `setUseDesktopUserAgent` is used only here; 

I moved that call to `currentTab.SetOverrideUserAgent()` for code cleanup, but of course you can leave it there

> but I think we should preserve the reloadOnChange behavior

yes, it' so, see line 262

> I am assuming that it is an optimisation used because native pages do not need to reload when agent changes

yes are java ui

> No need to pass an extra parameter here, it can be checked in the plumbing `SetOverrideUserAgentForAllTabs`/`SetOverrideUserAgent`

yes, absolutely not necessary, do you want me to change it?

Reply:
The parameter is not there, so there is nothing to change.